### ISSUE TITLE
Mullvad VPN GUI crashes due to Electro 26

### ISSUE THREAD

**Issue by Project579** [mullvad/mullvadvpn-app#5561]
## Issue report

Operating system: Arch Linux

App version: 2023.6

### Issue description

After the latest update to version 2023.6 the Mullvad VPN GUI application for Linux crashes after a few seconds, it seems to be caused by a bug in Electron 26: https://github.com/electron/electron/issues/32760

### Application log

```
[2023-12-07 00:34:55.807][verbose] Chromium sandbox is disabled
[2023-12-07 00:34:55.807][info] Running version 2023.6
[2023-12-07 00:34:55.894][info] Detected locale: en-US
[2023-12-07 00:34:55.941][info] Connected to the daemon
[2023-12-07 00:34:56.070][info] Skip autoconnect because GUI setting is disabled
[28098:1207/003458.465706:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[28098:1207/003458.478125:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[28098:1207/003500.270665:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[2023-12-07 00:35:01.469][error] Render process exited with exit code 139 due to crashed
[2023-12-07 00:35:01.469][info] before-quit received
[2023-12-07 00:35:01.470][info] Quit initiated
[2023-12-07 00:35:01.471][info] Unsubscribed from the daemon events
[2023-12-07 00:35:01.471][verbose] GRPC Channel connectivity state changed to 4
[2023-12-07 00:35:01.472][info] Disposable logging outputs disposed
[2023-12-07 00:35:01.472][info] Quit preparations finished
[2023-12-07 00:35:01.478][info] will-quit received
[2023-12-07 00:35:01.479][info] quit received
```

**Comment by raksooo** [comment]
Thanks for reporting! Is this happening every time you launch the GUI? Does it help if you launch it with `--disable-gpu`? (`/opt/Mullvad\ VPN/mullvad-vpn --disable-gpu`)

**Comment by m2rk1312** [comment]
Having the same issue. --disable-gpu didn't fix it for me, still crashing when you hover over the application

**Comment by ADHDefy** [comment]
Having the same issue on Fedora 39 after the newest update, though for me, the `--disable-gpu` option does *seem* to fix it. It's at least more stable for now. I'll update if it crashes again.

**Comment by mav-noob** [comment]
Same under Linux Mint 21.2 / Base: Ubuntu 22.04.3 LTS (Jammy)

App version: 2023.6

And also --disable-gpu doesn't solve the problem.

Any progress here?

**Comment by SyntheticBird45** [comment]
Hi. Arch linux user here.
App version: 2023.6
`--disable-gpu` doesn't help

logs:
```
$ mullvad-vpn --disable-gpu
[2023-12-xx 00:30:34.816][verbose] Chromium sandbox is disabled
[2023-12-xx 00:30:34.817][info] Running version 2023.6
[2023-12-xx 00:30:34.879][info] Detected locale: en
[2023-12-xx 00:30:34.914][info] Connected to the daemon
[2023-12-xx 00:30:34.970][info] Autoconnect the tunnel
[19686:1221/140837.678201:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[19686:1221/140839.141300:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[2023-12-xx 00:30:43.144][error] Render process exited with exit code 139 due to crashed
[2023-12-xx 00:30:43.144][info] before-quit received
[2023-12-xx 00:30:43.144][info] Quit initiated
[2023-12-xx 00:30:43.144][info] Unsubscribed from the daemon events
[2023-12-xx 00:30:43.145][verbose] GRPC Channel connectivity state changed to 4
[2023-12-xx 00:30:43.145][info] Disposable logging outputs disposed
[2023-12-xx 00:30:43.145][info] Quit preparations finished
[2023-12-xx 00:30:43.149][info] will-quit received
[2023-12-xx 00:30:43.149][info] quit received
```

**Comment by lasumn** [comment]
NixOS user here, GUI starts normally and I could input my Account Number but any further interaction crashes the GUI. --disable-gpu actually worked for me!

**Comment by schweineimweltall** [comment]
> Having the same issue on Fedora 39 after the newest update, though for me, the `--disable-gpu` option does _seem_ to fix it. It's at least more stable for now. I'll update if it crashes again.

Confirming: '--disable-gpu' is also working for me on Fedora 39

**Comment by Jo-Panic** [comment]
Fedora 39 + Gnome (Wayland).
Mullvad 2023.6

I don't know if my issue has same origin but the mullvad window displays nothing (see screenshot). Launching with --disable-gpu doesn't fix the issue.
![MullvadGUI-24-12-2023-Fedora39](https://github.com/mullvad/mullvadvpn-app/assets/95019115/050d0b31-df72-4d4d-9178-71efeadeea27)

Even if the window appears empty, it responds to certain mouse-click interactions, but no elements are displayed.

**Comment by firefoxlover** [comment]
Yes for me too the latest upgrade is a huge regression. But this is another bug. Maybe relevant is that my GUI is completely pixelated mess, nothing usable at all.

**Comment by twann4** [comment]
Same issue on Artix Linux

**Comment by seekermoc** [comment]
I'm using Fedora 39 Cinnamon on X11, but am seeing the same as @Jo-Panic above since installing 2023.6. When opened, the app window is either all white, all black, or black on the top third of the window and blue on the bottom (like it would look normally, but none of the text / buttons / etc appear).

The app doesn't crash or freeze, clicking on the window title bar close button still closes it, but it appears the GUI doesn't load or render completely. The VPN still works, the tray icon and its context menu controls (reconnect, disconnect, quit) still work, but the app window / GUI itself doesn't.

Running it with --disable-gpu _does_ fix it for me. If it doesn't work for you, make sure you "disconnect & quit" the currently running GUI via the tray icon before you try opening it again with --disable-gpu.

**Comment by Project579** [comment]
I have managed to solve the issue on my end without the use of `--disable-gpu`, I upgraded _mesa_ to version _23.3.2_ (latest on ArchLinux as of this writing), deleted any config files regarding Mullvad VPN in my home folder and that seems to have solved the issue, currently it functions just fine; I'm also running the app using the Wayland backed in case that makes a difference, also as another possible solution that has worked for me in the past with other applications is deleting the mesa shaders cache in your users home, as that can also sometimes cause weird crashes when mesa receives breaking changes.

Also I'm using an AMD GPU in case that makes any difference.

**Comment by BEaXt7f97** [comment]
Same issue for me MacOS Sonoma 2023.6

**Comment by Tre-brock** [comment]
Same issue on linux mint

**Comment by TJonCooper** [comment]
Same on Ubuntu 22.04.3 LTS (kernel = 6.5). It typically happens after waking my computer, but can happen at anytime. I'm using X11.

**Comment by MarkusPettersson98** [comment]
@Project579 Have you had the GUI crash on you lately? We have bumped Electron quite a few times since.

**Comment by Wotts** [comment]
@MarkusPettersson98 Hi there, for me on Ubuntu 24.04 the GUI is sometimes crashing on login. It does restart itself though.
`--disable-gpu` makes no difference.
There's a few errors in `/var/log/mullvad-vpn/daemon.log` however I doubt if those are an issue and if they are relevant to the GUI.

I've hit the Send button on the Crash Report Ubuntu presents, so hopefully you've received those. Otherwise I can share the last crash report with you privately.

**Comment by MarkusPettersson98** [comment]
@Wotts Please do share the last crash report with us! You can do so by contacting support*, referencing this issue and that it should be forwarded to me :)

Thanks<3

*https://mullvad.net/en/blog/we-now-self-host-our-support-email

**Comment by Wotts** [comment]
Hey @MarkusPettersson98 I installed `mullvad-vpn 2026.2` but these crashes still occur.
It's pretty consistently every time shortly after logging in to Ubuntu - but only once.

I presume you didn't find anything in the crash log I sent. I did stop hitting the 'send crash report' button a while ago as it became a bit tedious ;)

Is there anything else I can test or check for you guys? I'd love to help out!

**Comment by tobias-jarvelov** [comment]
@Wotts We unfortunately do not receive the crash reports sent by the Ubuntu dialog. Those are only sent to Ubuntu and their maintainers. In order to share the logs with us you must do the following after you encounter a crash next time:

1. Open the app again
2. Go into Settings -> Support -> Report a problem
3. Add a link to this Github issue in the message field

Thanks!

**Comment by Wotts** [comment]
@tobias-jarvelov perfect, I just did that this morning :)
